src/main/java/org/skriptlang/skript/bukkit/recipes/elements/ExprSecCreateRecipe.java

```diff
@@ -0,0 +1,194 @@
+		Skript.registerExpression(ExprSecCreateRecipe.class, Recipe.class, ExpressionType.SIMPLE,
+			"a new %*recipetype% with [the] (key|id) %string%");
+		EventValues.registerEventValue(CreateRecipeEvent.class, Recipe.class, CreateRecipeEvent::getMutableRecipe);
+	}
+
+	private RecipeType providedType;
+	private Expression<String> providedName;
+	private Trigger trigger;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int pattern, Kleenean isDelayed, ParseResult result, @Nullable SectionNode node, @Nullable List<TriggerItem> triggerItems) {
+		if (node == null) {
+			Skript.error("Creating a new recipe requires a section.");
+			return false;
+		}
+		//noinspection unchecked
+		providedType = ((Literal<RecipeType>) exprs[0]).getSingle();
+		if (providedType == RecipeType.COOKING || providedType == RecipeType.CRAFTING || providedType == RecipeType.COMPLEX) {
+			Skript.error("You can not create a new '" + providedType + "' recipe type.");
+			return false;
+		} else if (providedType == RecipeType.SMITHING && !SUPPORT_SMITHING) {
+			Skript.error("You can not create a new 'smithing' recipe type on MC version 1.20+.");
+			return false;
+		} else if (providedType.getRecipeClass() == null) {
+			Skript.error("You can not create a new '" + providedType + "' recipe type on this MC version.");
+			return false;
+		}
+		//noinspection unchecked
+		providedName = (Expression<String>) exprs[1];
+		AtomicBoolean delayed = new AtomicBoolean(false);
+		Runnable afterLoading = () -> delayed.set(!getParser().getHasDelayBefore().isFalse());
+		//noinspection unchecked
+		trigger = loadCode(node, "create recipe", afterLoading, providedType.getEventClass());
+		if (delayed.get()) {
+			Skript.error("Delays cannot be used within a 'create recipe' section.");
+			return false;
+		}
+		return true;
+	}
+
+	@Override
+	protected Recipe @Nullable [] get(Event event) {
+		String name = providedName.getSingle(event);
+		if (name == null || name.isEmpty()) {
+			//Skript.error("The id for a recipe must not be null nor empty.");
+			return null;
+		}
+		NamespacedKey key = NamespacedKey.fromString(name, Skript.getInstance());
+		if (key == null) {
+			//Skript.error("The provided id is invalid.")
+			return null;
+		}
+		CreateRecipeEvent recipeEvent = switch (providedType) {
+			case SHAPED -> new ShapedRecipeEvent(key);
+			case SHAPELESS -> new ShapelessRecipeEvent(key);
+			case BLASTING -> new BlastingRecipeEvent(key);
+			case FURNACE -> new FurnaceRecipeEvent(key);
+			case CAMPFIRE -> new CampfireRecipeEvent(key);
+			case SMOKING -> new SmokingRecipeEvent(key);
+			case SMITHING -> new SmithingRecipeEvent(key, providedType);
+			case SMITHING_TRANSFORM -> new SmithingTransformRecipeEvent(key);
+			case SMITHING_TRIM -> new SmithingTrimRecipeEvent(key);
+			case STONECUTTING -> new StonecuttingRecipeEvent(key);
+			case TRANSMUTE -> new TransmuteRecipeEvent(key);
+			default -> throw new IllegalStateException("Unexpected value: " + providedType);
```

Comment:
It might be worthwile to delegate this behavior to the enum, what do you think of:
```java
public @Nullable CreateRecipeEvent createRecipeEvent(NamespacedKey key, RecipeType providedType) {
    return recipeEventFunction.apply(key, providedType);
}
```
where enums would specify the `recipeEventFunction`.

Could be applicable to other switch cases in the same regard